### PR TITLE
all users | authorization | navigation bar

### PR DESCRIPTION
**Pull request by mla-wtag**
[JTJ-33](https://welltravel.atlassian.net/jira/software/projects/JTJ/boards/44?selectedIssue=JTJ-33)

**Description**
- In this PR, I primarily addressed some link issues beside adding a index view.
- The "task" link in the navigation bar hasn't been implemented yet for this reason there's a temporary link to the root_path.
 

https://github.com/mla-wtag/ml_Journey/assets/141820916/5705d44d-39cf-4e71-b29c-8d5432dba724

### CODE REVIEW

File: app/controllers/user_sessions_controller.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 class UserSessionsController < ApplicationController
+  include ApplicationHelper
```
Comment (chm-wtag) [inline app/controllers/user_sessions_controller.rb]: it should work even without this line, could you please try? 🤔

**Review by chm-wtag — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -4,4 +4,9 @@ class ApplicationController < ActionController::Base
   def current_user
     @current_user ||= session[:user_id] && User.find_by(id: session[:user_id])
   end
+
+  rescue_from CanCan::AccessDenied do |exception|
+    flash[:alert] = t('alerts.not_permitted')
+    redirect_to user_path(current_user)
+  end
```
Comment (evk-wtag) [inline app/controllers/application_controller.rb:8]: Let's rearrange it and place it above the current_user method for readability

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    <% @user.each do |user| %> 
+     <% next if user == current_user || !user.confirmed? %> 
+      <tr>
+        <td><%= user.first_name %></td>
+        <br>
+        <td><%= user.last_name %></td>
+        <br>
+        <td><%= user.role %></td>
+        <br>
+          <td><%= link_to t('views.profile'), user_path(user) %></td>
```
Comment (evk-wtag) [inline app/views/users/index.html.erb:19]:
```suggestion
        <td><%= link_to t('views.profile'), user_path(user) %></td>
```

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        <br>
```
Comment (evk-wtag) [inline app/views/users/index.html.erb]:
```suggestion
        <br>
        <td><%= link_to t('views.profile'), user_path(user) %></td>
        <br>
```

File: spec/controllers/users_controller_spec.rb [hunk 5]
```diff
@@ -3,6 +3,29 @@
   let(:user) { FactoryBot.create(:user) }
   before { sign_in(user) }
 
+  describe 'GET #index' do
+    context 'when user has admin role' do
+      let(:user) { FactoryBot.create(:user, role: 'admin_role') }
```
Comment (evk-wtag) [inline spec/controllers/users_controller_spec.rb]:
```suggestion
      let(:user) { FactoryBot.create(:user, role: :admin_role) }
```

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -3,6 +3,29 @@
...
+      before { sign_in(user) }
+
+      it 'renders the :index template' do
+        get :index
+        expect(response).to render_template(:index)
+      end
+    end
+
+    context 'when user has regular role' do
+      let(:user) { FactoryBot.create(:user, role: 'user_role') }
```
Comment (evk-wtag) [inline spec/controllers/users_controller_spec.rb]:
```suggestion
      let(:user) { FactoryBot.create(:user, role: :user_role) }
```

**Review by evk-wtag — APPROVED** [review]

File: app/controllers/user_sessions_controller.rb (hunk 1 above)
Comment (mla-wtag) [reply app/controllers/user_sessions_controller.rb]: I removed it from users_controller.rb and user_sessions_controller.rb, but added the line to application_controller.rb

File: app/views/layouts/_navbar.html.erb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
```
Comment (lra-wtag) [inline app/views/layouts/_navbar.html.erb]: i18n please

**Review by lra-wtag — CHANGES_REQUESTED** [review]

**Review by lra-wtag — APPROVED** [review]

**Pull request merged**